packages/ai-core/src/artifact-api.test.ts

```diff
@@ -0,0 +1,76 @@
+import nock from 'nock';
+import { HttpDestination } from '@sap-cloud-sdk/connectivity';
+import {
+  ArtifactApi,
+  ArtifactCreationResponse,
+  ArtifactList,
+  ArtifactPostData
+} from './index.js';
+
+describe('artifact', () => {
+  const destination: HttpDestination = {
+    url: 'https://ai.example.com'
+  };
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('get artifact parses a successful response', async () => {
+    const expectedResponse: ArtifactList = {
+      count: 1,
+      resources: [
+        {
+          createdAt: '2024-07-08T13:36:41Z',
+          description: 'dataset for training test',
+          id: '0a1b2c3d',
+          kind: 'dataset',
+          modifiedAt: '2024-07-08T13:36:41Z',
+          name: 'training-test-data',
+          scenarioId: 'foundation-models',
+          url: 'ai://default/spam/data'
+        }
+      ]
+    };
+
+    nock(destination.url).get('/lm/artifacts').reply(200, expectedResponse, {
+      'Content-Type': 'application/json',
+      'AI-Resource-Group': 'default'
```

Comment:
[q] Aren't these part of the request header?

Reply:
You're right. The resource group is only a part of the request header. Only content type should be a part of response. Thanks for catching it.

Reply:
I would also not bother so much with response headers for now.